problem.go

```diff
@@ -0,0 +1,98 @@
+package main
+
+import (
+	"math/rand"
+	"fmt"
+	"strconv"
+	"strings"
+	"errors"
+)
+
+type Problem struct {
+	problem string
+	solution string
+}
+
+func (p *Problem) checkSolution(solution string) bool {
+	pis, err := strconv.Atoi(p.solution)
+	if err != nil { panic(fmt.Sprintf("Invalid solution '%s'", p.solution)) }
+
+	r := strings.NewReplacer(" ", "", "\t", "", "\n", "");
+	is, err := strconv.Atoi(r.Replace(solution))
+	if err != nil {
+		 return false
+	}
+	return is == pis
+}
+
+func ProblemForClient(c *Client) *Problem {
```

Comment:
```suggestion
func Problem(c *Client) *Problem {
```
Don't be verbose, client is already the argument